x/incentives/keeper/store.go

```diff
@@ -114,8 +114,28 @@ func (k Keeper) SetGroup(ctx sdk.Context, group types.Group) {
 }
 
 // GetAllGroupGauges gets all the groupGauges that is in state.
-func (k Keeper) GetAllGroups(ctx sdk.Context) ([]types.Group, error) {
-	return osmoutils.GatherValuesFromStorePrefix(ctx.KVStore(k.storeKey), types.KeyPrefixGroup, k.ParseGroupFromBz)
+func (k Keeper) GetAllGroups(ctx sdk.Context) []types.Group {
+	groups, err := osmoutils.GatherValuesFromStorePrefix(ctx.KVStore(k.storeKey), types.KeyPrefixGroup, k.ParseGroupFromBz)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Let's please continue returning an error. We are panicking in the AfterEpochEnd hook which seems concerning even if the panics are caught somewhere above the call stack

Reply:
Returning error here https://github.com/osmosis-labs/osmosis/pull/6565/commits/f25c6c6f1479a266769ed70db44ec90b58a504f7